hawkbit-security-integration/src/main/java/org/eclipse/hawkbit/security/PreAuthTokenSourceTrustAuthenticationProvider.java

```diff
@@ -118,6 +110,36 @@ public Authentication authenticate(final Authentication authentication) {
 
         throw new BadCredentialsException("The provided principal and credentials are not match");
     }
+    
+	/**
+	 * 
+	 * The credentials may either be of type HeaderAuthentication or of type
+	 * Collection<HeaderAuthentication> depending on the authentication mode in
+	 * use (the latter is used in case of trusted reverse-proxy). It is checked
+	 * whether principal equals credentials (respectively if credentials
+	 * contains principal in case of collection) because we want to check if
+	 * e.g. controllerId containing in the URL equals the controllerId in the
+	 * special header set by the reverse-proxy which extracted the CN from the
+	 * certificate.
+	 * 
+	 * @param principal
+	 * @param credentials
+	 * @param tokenDetails
+	 * @return
+	 */
+    private boolean calculateAuthenticationSuccess(Object principal, Object credentials, Object tokenDetails) {
+    	boolean successAuthentication = false;
+    	if (principal.equals(credentials)) {
```

Comment:
I would recommend switching the if-else statement because in case of a `Collection` we do not need to try if they are equal `principal.equals(credentials)`. I would probably to make sense to check first of `Collection` and do the equal collection check and in the else do the actual `principal.equals(credentials)` or is there another case I'm missing.